src/tools/map/xml/creator/MapXmlUIHelper.java

```diff
@@ -162,18 +162,16 @@ public static GridBagConstraints getGBCCloneWith(final GridBagConstraints gbcToC
   }
 
   public static FileOpen selectFile(final String filename, final File currentDirectory, final String... extensions) {
-    do {
-      final FileOpen gameXmlFileOpen =
-          new FileOpen("Load " + filename, currentDirectory, MapXmlCreator.FILE_NAME_ENDING_XML);
-      if (gameXmlFileOpen.getPathString() == null) {
-        JOptionPane.showMessageDialog(null,
-            "No valid selection for " + filename + "!",
-            "Error while selecting " + filename, JOptionPane.ERROR_MESSAGE);
-      } else {
-        return gameXmlFileOpen;
-      }
-    } while (1 > 0);
-
+    final FileOpen gameXmlFileOpen =
+        new FileOpen("Load " + filename, currentDirectory, MapXmlCreator.FILE_NAME_ENDING_XML);
+    if (gameXmlFileOpen.getPathString() != null) {
```

Comment:
If the window gets closed the gameXMLFileOpen.getPathString() is null.
In this case A error message is shown, and the window is closed after clicking ok